File: ios/Sources/Common/Export/CurlExporter.swift
import Foundation

/// Generates cURL command strings from captured network requests.
public enum CurlExporter {
    /// Generate a cURL command for the given request.
    public static func export(_ request: NetworkRequest) -> String {
        var parts = ["curl"]

        if request.method != .get {
            parts.append("-X \(request.method.rawValue)")
        }

        for (key, values) in request.requestHeaders.sorted(by: { $0.key < $1.key }) {
            for value in values {
                let escapedKey = key.replacingOccurrences(of: "'", with: "'\\''")
                let escaped = value.replacingOccurrences(of: "'", with: "'\\''")
                parts.append("-H '\(escapedKey): \(escaped)'")
            }
        }

        if let body = request.requestBody {
            let escaped = body.replacingOccurrences(of: "'", with: "'\\''")
            parts.append("-d '\(escaped)'")
        }

        let urlEscaped = request.url.replacingOccurrences(of: "'", with: "'\\''")
        parts.append("'\(urlEscaped)'")

        return parts.joined(separator: " \\\n  ")
    }
}

File: android/hakka-network/src/main/kotlin/com/noodleapps/hakka/export/CurlExporter.kt
package com.noodleapps.hakka.export

import com.noodleapps.hakka.NetworkRequest

/**
 * Generates cURL command strings from [NetworkRequest] objects.
 */
object CurlExporter {

    /** Generates a single-line cURL command for the given request. */
    fun export(request: NetworkRequest): String = buildString {
        append("curl")
        if (request.method.name != "GET") {
            append(" -X ").append(request.method.name)
        }
        for ((name, values) in request.requestHeaders) {
            for (value in values) {
                append(" -H '").append(escapeShell(name)).append(": ").append(escapeShell(value)).append("'")
            }
        }
        request.requestBody?.let { body ->
            append(" -d '").append(escapeShell(body)).append("'")
        }
        append(" '").append(escapeShell(request.url)).append("'")
    }

    /** Escapes a string for safe inclusion inside single-quoted shell arguments. */
    private fun escapeShell(value: String): String = value.replace("'", "'\"'\"'")
}
